### PR TITLE
Added new controls for restricting Admin/User access on Agentpool and…

### PR DESCRIPTION
**Pull request by SaikumarRagala**
… svcCOnn

## Description
</br>
Added the controls to restrict User/Administrator access  for broader groups on agentpool/serviceconnection.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.AgentPool.json [hunk 1]
```diff
@@ -156,6 +156,23 @@
         "DP"
       ],
       "Enabled": true
+    },
+    {
+      "ControlID": "ADO_Agentpool_Review_Broader_Group_Access",
+      "Description": "Ensure that broader groups has limited access(Reader) to the agentpool.",
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.AgentPool.json:162]: space in agent and pool

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 2]
```diff
@@ -317,55 +317,98 @@ class AgentPool: ADOSVTBase
...
         $this.agentPoolActivityDetail.isComputed = $true
     }
+
+    hidden [ControlResult] CheckBroaderGroupAccess ([ControlResult] $controlResult) {
+        $failMsg = $null
+        try {
+            $restrictedGroups = @();
+
+            if ($this.ControlSettings -and [Helpers]::CheckMember($this.ControlSettings, "AgentPool.RestrictedBroaderGroupsForAgentpool") ) {
+                $restrictedGlobalGroupsForSerConn = $this.ControlSettings.AgentPool.RestrictedBroaderGroupsForAgentpool;
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:378]: rename this 'restrictedGlobalGroupsForSerConn' variable itz for agentpools

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 3]
```diff
@@ -317,55 +317,98 @@ class AgentPool: ADOSVTBase
...
+                if ((($this.AgentObj | Measure-Object).Count -gt 0) -and [Helpers]::CheckMember($this.AgentObj, "identity")) {
+                    # match all the identities added on agentpool with defined restricted list
+                    $roleAssignments = @();
+                    $roleAssignments +=   ($this.AgentObj | Select-Object -Property @{Name="Name"; Expression = {$_.identity.displayName}},@{Name="Role"; Expression = {$_.role.displayName}});
+                    $restrictedGroups = $roleAssignments | Where-Object { $restrictedGlobalGroupsForSerConn -contains $_.Name.split('\')[-1] -and ($_.Role -eq "Administrator" -or $_.Role -eq "User") }
+
+                    # fail the control if restricted group found on agentpool
+                    if ($restrictedGroups) {
+                        $controlResult.AddMessage("Total number of global groups that have Administration/User access to agentpool: ", ($restrictedGroups | Measure-Object).Count)
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:387]: Administration or Administrator?

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 4]
```diff
@@ -317,55 +317,98 @@ class AgentPool: ADOSVTBase
...
+                        $controlResult.AddMessage([VerificationResult]::Failed, "Do not grant global groups Administration/User access to agentpools. Granting elevated permissions to these groups can risk exposure of agentpools to unwarranted individuals.");
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:388]: Agent pools should be space

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 5]
```diff
@@ -317,55 +317,98 @@ class AgentPool: ADOSVTBase
...
+            else {
+                $controlResult.AddMessage([VerificationResult]::Manual, "List of restricted global groups for agentpool is not defined in your organization policy. Please update your ControlSettings.json as per the latest AzSK.ADO PowerShell module.");
+            }
+        }
+        catch {
+            $failMsg = $_
+        }
+
+        if (![string]::IsNullOrEmpty($failMsg)) {
+            $controlResult.AddMessage([VerificationResult]::Manual, "Unable to fetch agentpools details. $($failMsg)Please verify from portal that you are not granting global security groups access to agentpools");
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:410]: Manual or Error as it is coming if  some exception occurred flow goes to catch

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 2 above)
Comment (SaikumarRagala) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: Addressed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 3 above)
Comment (SaikumarRagala) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: Addressed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 4 above)
Comment (SaikumarRagala) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: Addressed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 5 above)
Comment (SaikumarRagala) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: Addressed

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**